agent/hcp/telemetry/gauge_store.go

```diff
@@ -0,0 +1,61 @@
+package telemetry
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/instrument"
+)
+
+type gaugeStore struct {
```

Comment:
Might be good to add some comments here as to why we need a gaugeStore and we don't need a counterStore/histogramStore

Reply:
Added one, let me know if it can be improved!